### PR TITLE
Allows Prometheans to be cloned from their Core.

### PR DESCRIPTION
**Pull request by Mechoid**
Brains can be set to be a source of genetic information for cloning scanners. Currently only enabled on Promethean cores.

Promethean cores are now 'open containers', to allow addition / removal of chemicals, currently has no purpose, but will in Take 2.

Allows species to have a special cloning modifier, only Prometheans currently have one.

Fixes a spelling mistake in the cloning modifiers file.

### CODE REVIEW

File: code/game/dna/dna_modifier.dm [hunk 1]
```diff
@@ -53,6 +53,7 @@
 	var/mob/living/carbon/occupant = null
 	var/obj/item/weapon/reagent_containers/glass/beaker = null
 	var/opened = 0
+	var/allowed_braintype = /obj/item/organ/internal/brain/slime //Incase an admin wishes to varchange this to allow another type of brain, or subtypes are needed.
```
Comment (Anewbe) [inline code/game/dna/dna_modifier.dm]: Maybe toss this var on the brain instead, so if admins specifically want something to be brain-cloneable they don't have to touch the machine with pathing?

File: code/game/dna/dna_modifier.dm [hunk 2]
```diff
@@ -134,6 +134,22 @@
 		item.loc = src
 		user.visible_message("\The [user] adds \a [item] to \the [src]!", "You add \a [item] to \the [src]!")
 		return
+
+	else if(istype(item, /obj/item/organ/internal/brain))
+		if (src.occupant)
+			user << "<span class='warning'>The scanner is already occupied!</span>"
```
Comment (Anewbe) [inline code/game/dna/dna_modifier.dm]: to_chat

File: code/game/dna/dna_modifier.dm [hunk 3]
```diff
@@ -134,6 +134,22 @@
...
+			return
+		var/obj/item/organ/internal/brain/brain = item
+		if(brain.clone_source)
+			user.drop_item()
+			brain.loc = src
+			put_in(brain.brainmob)
```
Comment (Anewbe) [inline code/game/dna/dna_modifier.dm]: I assume this makes the brainmob the occupant?

Comment (Mechoid) [reply code/game/dna/dna_modifier.dm]: It does.

**Pull request merged**